plugins/inputs/prometheus/prometheus.go

```diff
@@ -422,6 +491,7 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 
 	var err error
 	var resp *http.Response
+	start := time.Now()
```

Comment:
This should be right before the client.Do and the time captured immediately after the client.Do. That way we avoid the if statements + error checking getting included in the total time.